src/state/recordedState.ts

```diff
@@ -184,10 +184,6 @@ export class RecordedState {
 
   public hasRunOperator = false;
 
-  public hasRunSurround = false;
```

Comment:
Don't have time yet for a thorough review, but I love seeing jank like this being ripped out!